src/App.js

```diff
@@ -1,40 +1,84 @@
-import { useState } from 'react';
+import { useState, useMemo } from 'react';
 import { Row, Divider, Button } from 'antd';
 import './App.css';
-import foods from "./foods.json";
+import foods from './foods.json';
 import FoodBox from './components/FoodBox';
-
+import AddFoodForm from './components/AddFoodForm';
+import Search from './components/Search';
 
 function App() {
-  const [food, setFood] = useState(foods)
-  return (
-    <div className="App">
+  const [foodArr, setFoodArr] = useState(foods);
+  const [showFood, setShowFood] = useState(foods);
+  const [visible, setVisible] = useState(false);
+
+  const addFood = (food) => {
+    const foodArrCopy = [...foodArr, food];
+    setFoodArr(foodArrCopy);
+    setShowFood(foodArrCopy);
+  };
 
-      <ul>
-        {food.map((e) => {
-          console.log(e)
+  const searchFood = (query) => {
+    const filteredFood = foodArr.filter((food) => {
+      return food.name.toLowerCase().includes(query.toLowerCase());
+    });
+    setShowFood(filteredFood);
+  };
+
+  const deleteFood = (foodName) => {
+    const filteredFood = foodArr.filter((food) => {
+      return food.name !== foodName;
+    });
+    setFoodArr(filteredFood);
+    setShowFood(filteredFood);
+  };
+  
+  const toggleFormVisibility = () => {setVisible(!visible)};
+
+//---------------------------------Tried useMemo to solve iteration 8
+/*   const displayFood = useMemo(
+    () =>
+    {
+        if(foodArr.length === 0) return <div>EMPTY</div>
+        showFood.map((food) => {
           return (
-            <li>
-              <div>
-                <FoodBox food={e}/>
-              </div>
-            </li>
-          )
-        })}
-      </ul>
-      {/* Display Add Food component here */}
+            <FoodBox
+              span={6}
+              food={food}
+              deleteFood={deleteFood}
+              key={food.name}
+            />
+          );
+        })
+    },
+    [foodArr, deleteFood, showFood]
+) */
 
-      <Button> Hide Form / Add New Food </Button>
 
-      {/* Display Search component here */}
+  return (
+    <div className="App">
+      
+      <Button onClick={toggleFormVisibility}>{visible ? "Hide Form" : "Add New Food"}</Button>
+      {visible && <AddFoodForm addFood={addFood} />}
+
+      <Search searchFood={searchFood} />
 
       <Divider>Food List</Divider>
 
-      <Row style={{ width: '100%', justifyContent: 'center' }}>
-        {/* Render the list of Food Box components here */}
+      <Row style={{ width: '100%', justifyContent: 'center' }} gutter={[16, 16]}>
+        {showFood.map((food) => {
+          return (
+            <FoodBox
+              span={6}
+              food={food}
+              deleteFood={deleteFood}
+              key={food.name}
+            />
+          );
+        })}
       </Row>
+
```

Comment:
Iteration 8 - if you want to show something when there are no foods left, and foods is an array, you can check the length of the array:

```
function App() {

  return (
    <div className="App">
                    (...)
      { !showFood.length && 
        <p>No results found!</p> }
    </div>
)}
```